drivers/usb/device/usb_dc_stm32.c

```diff
@@ -194,88 +196,55 @@ void HAL_PCD_SOFCallback(PCD_HandleTypeDef *hpcd)
 	 * that instead.  Example reference manual RM0360 for
 	 * STM32F030x4/x6/x8/xC and STM32F070x6/xB.
 	 */
-#if defined(RCC_HSI48_SUPPORT) || \
-	defined(CONFIG_SOC_SERIES_STM32WBX) || \
-	defined(CONFIG_SOC_SERIES_STM32H7X) || \
-	defined(CONFIG_SOC_SERIES_STM32L5X) || \
-	defined(CONFIG_SOC_SERIES_STM32U5X)
-#if !STM32_HSI48_ENABLED
+
+#if !STM32_HSI48_ENABLED && !LL_RCC_USB_CLKSOURCE_NONE
 	/* Deprecated: enable HSI48 using device tree */
 #warning USB device requires HSI48 clock to be enabled using device tree
 #endif /* ! STM32_HSI48_ENABLED*/
-	/*
-	 * In STM32L0 series, HSI48 requires VREFINT and its buffer
-	 * with 48 MHz RC to be enabled.
-	 * See ENREF_HSI48 in reference manual RM0367 section10.2.3:
-	 * "Reference control and status register (SYSCFG_CFGR3)"
-	 */
-#ifdef CONFIG_SOC_SERIES_STM32L0X
-	if (LL_APB2_GRP1_IsEnabledClock(LL_APB2_GRP1_PERIPH_SYSCFG)) {
-		LL_SYSCFG_VREFINT_EnableHSI48();
-	} else {
-		LOG_ERR("System Configuration Controller clock is "
-			"disabled. Unable to enable VREFINT which "
-			"is required by HSI48.");
-	}
-#endif /* CONFIG_SOC_SERIES_STM32L0X */
-
-	z_stm32_hsem_lock(CFG_HW_CLK48_CONFIG_SEMID, HSEM_LOCK_DEFAULT_RETRY);
-
-	/* Keeping this sequence for legacy: */
-	LL_RCC_HSI48_Enable();
-	while (!LL_RCC_HSI48_IsReady()) {
-		/* Wait for HSI48 to become ready */
-	}
-
-	LL_RCC_SetUSBClockSource(LL_RCC_USB_CLKSOURCE_HSI48);
 
 #ifdef CONFIG_SOC_SERIES_STM32U5X
 	/* VDDUSB independent USB supply (PWR clock is on) */
 	LL_PWR_EnableVDDUSB();
 #endif /* CONFIG_SOC_SERIES_STM32U5X */
 
-#if !defined(CONFIG_SOC_SERIES_STM32WBX)
-	/* Specially for STM32WB, don't unlock the HSEM to prevent M0 core
-	 * to disable HSI48 clock used for RNG.
-	 */
-	z_stm32_hsem_unlock(CFG_HW_CLK48_CONFIG_SEMID);
-#endif /* CONFIG_SOC_SERIES_STM32WBX */
+#if (DT_INST_NUM_CLOCKS(0) == 1) & (defined(CONFIG_SOC_SERIES_STM32F0X) || \
+					defined(CONFIG_SOC_SERIES_STM32G0X) || \
+					defined(CONFIG_SOC_SERIES_STM32L0X) || \
+					defined(CONFIG_SOC_SERIES_STM32H7X))
+#warning USB domain clock not configured
+#endif
 
-#elif defined(LL_RCC_USB_CLKSOURCE_NONE)
+#if defined(LL_RCC_USB_CLKSOURCE_NONE)
 	/* When MSI is configured in PLL mode with a 32.768 kHz clock source,
 	 * the MSI frequency can be automatically trimmed by hardware to reach
 	 * better than ±0.25% accuracy. In this mode the MSI can feed the USB
 	 * device. For now, we only use MSI for USB if not already used as
 	 * system clock source.
 	 */
-#if STM32_MSI_PLL_MODE && !STM32_SYSCLK_SRC_MSI
-	LL_RCC_MSI_Enable();
-	while (!LL_RCC_MSI_IsReady()) {
-		/* Wait for MSI to become ready */
-	}
-	/* Force 48 MHz mode */
-	LL_RCC_MSI_EnableRangeSelection();
-	LL_RCC_MSI_SetRange(LL_RCC_MSIRANGE_11);
-	LL_RCC_SetUSBClockSource(LL_RCC_USB_CLKSOURCE_MSI);
-#else
-	if (LL_RCC_PLL_IsReady()) {
-		LL_RCC_SetUSBClockSource(LL_RCC_USB_CLKSOURCE_PLL);
-	} else {
-		LOG_ERR("Unable to set USB clock source to PLL.");
+#warning TODO, configure MSI as USB clck source
+#endif
```

Comment:
@loicpoulain I'm having hard time to figure out which are the impacted references.
Since you authored these lines, can you tell ?

Reply:
What do you mean by references? The impacted boards? Need to double check that, afair trimmed MSI is used when no external crystal (hse) is connected to the microcontroller.

Reply:
Ok, thanks. I've tried to clarify things.